### ISSUE TITLE
Pandas parses csv file correctly but modin fails

### ISSUE THREAD

**Issue by thenewera-ru** [modin-project/modin#3146]
### System information
- **OS Platform and Distribution **: `Mac Os Mojave 10.14.6`
- **Modin version** `0.10.0`:
- **Python version**: `3.8.5`
- **Code we can use to reproduce**:

```
import ray
if ray.is_initialized():
    ray.shutdown()
    ray.init(address='auto')
else:
    ray.init()

from pathlib import Path
import modin.pandas as mpd
import pandas as pd

PATH = Path.home() / 'demo.csv'

# Pandas reads successfully with no problems
data = pd.read_csv(PATH, encoding='utf-8-sig', names=['Text', 'Annotation'], quotechar='"', skipinitialspace=True, sep=',', header=0)

# Modin throws an error 
data = mpd.read_csv(PATH, encoding='utf-8-sig', names=['Text', 'Annotation'], quotechar='"', skipinitialspace=True, sep=',', header=0)
```

### Describe the problem
As described in #1008 

### Source code / logs
```
  File "pandas/_libs/parsers.pyx", line 518, in pandas._libs.parsers.TextReader.__cinit__
  File "pandas/_libs/parsers.pyx", line 706, in pandas._libs.parsers.TextReader._get_header
  File "pandas/_libs/parsers.pyx", line 814, in pandas._libs.parsers.TextReader._tokenize_rows
  File "pandas/_libs/parsers.pyx", line 1951, in pandas._libs.parsers.raise_parser_error
pandas.errors.ParserError: Error tokenizing data. C error: EOF inside string starting at row 1
```

**Comment by thenewera-ru** [comment]
@datapythonista, To reproduce an error aboveI post this demo csv file. Pandas is ok with it but not modin, unfortunately. Hope this helps!
[demo.csv](https://github.com/modin-project/modin/files/6634781/demo.csv)

**Comment by devin-petersohn** [comment]
@thenewera-ru I can reproduce the issue with your CSV, it looks like it's an issue with the encoding not being properly handled in the workers, but it's still early to tell.

We'll get this worked on, thanks for reporting!

**Comment by vnlitvinov** [comment]
Fixed in [0.11.3](https://github.com/modin-project/modin/releases/tag/0.11.3)
